app/models/miq_schedule_worker/runner.rb

```diff
@@ -77,15 +77,19 @@ def schedules_for_all_roles
     # Schedule - Log current system configuration
     every = worker_settings[:log_active_configuration_interval]
 
-    scheduler.schedule_every(every, :tags => [:vmdb_appliance_log_config, schedule_category]) do
-      enqueue(:vmdb_appliance_log_config)
-    end
+    scheduler.schedule_every(
+      every,
+      :first_in => every,
+      :tags     => [:vmdb_appliance_log_config, schedule_category]
+    ) { enqueue(:vmdb_appliance_log_config) }
 
     # Schedule - Log current database statistics and bloat
     every = worker_settings[:log_database_statistics_interval]
-    scheduler.schedule_every(every, :tags => [:log_all_database_statistics, schedule_category]) do
-      enqueue(:vmdb_database_log_all_database_statistics)
-    end
+    scheduler.schedule_every(
+      every,
+      :first_in => every,
+      :tags     => [:log_all_database_statistics, schedule_category]
+    ) { enqueue(:vmdb_database_log_all_database_statistics) }
```

Comment:
Are there other places where we do a `every` without a `:first_in`?

Reply:
not that I can tell: 

```
13:13 $ gg -C1 "every," 
app/models/miq_schedule.rb-307-      _log.info("Schedule [#{name}] #{message} to run at #{time} with interval #{int}")
app/models/miq_schedule.rb:308:      options = {:method => :schedule_every, :interval => int, :schedule_id => id, :discard_past => true, :first_at => time, :tags => tag}
app/models/miq_schedule.rb-309-    end
--
app/models/miq_schedule_worker/runner.rb-80-    scheduler.schedule_every(
app/models/miq_schedule_worker/runner.rb:81:      every,
app/models/miq_schedule_worker/runner.rb:82:      :first_in => every,
app/models/miq_schedule_worker/runner.rb-83-      :tags     => [:vmdb_appliance_log_config, schedule_category]
--
app/models/miq_schedule_worker/runner.rb-88-    scheduler.schedule_every(
app/models/miq_schedule_worker/runner.rb:89:      every,
app/models/miq_schedule_worker/runner.rb:90:      :first_in => every,
app/models/miq_schedule_worker/runner.rb-91-      :tags     => [:log_all_database_statistics, schedule_category]
--
app/models/miq_schedule_worker/runner.rb-96-    scheduler.schedule_every(
app/models/miq_schedule_worker/runner.rb:97:      every,
app/models/miq_schedule_worker/runner.rb:98:      :first_in => every,
app/models/miq_schedule_worker/runner.rb-99-      :tags     => [:status_update, schedule_category]
--
app/models/miq_schedule_worker/runner.rb-104-    scheduler.schedule_every(
app/models/miq_schedule_worker/runner.rb:105:      every,
app/models/miq_schedule_worker/runner.rb:106:      :first_in => every,
app/models/miq_schedule_worker/runner.rb-107-      :tags     => [:log_status, schedule_category]
--
app/models/miq_schedule_worker/runner.rb-136-    every = worker_settings[:job_timeout_interval]
app/models/miq_schedule_worker/runner.rb:137:    scheduler.schedule_every(every, :first_in => every) do
app/models/miq_schedule_worker/runner.rb-138-      enqueue(:job_check_jobs_for_timeout)
--
app/models/miq_schedule_worker/runner.rb-143-    every = [worker_settings[:service_retired_interval], worker_settings[:vm_retired_interval], worker_settings[:orchestration_stack_retired_interval], worker_settings[:load_balancer_retired_interval]].min
app/models/miq_schedule_worker/runner.rb:144:    scheduler.schedule_every(every, :first_in => every) do
app/models/miq_schedule_worker/runner.rb-145-      enqueue(:retirement_check)
--
app/models/miq_schedule_worker/runner.rb-149-    every = worker_settings[:authentication_check_interval]
app/models/miq_schedule_worker/runner.rb:150:    scheduler.schedule_every(every, :first_in => every) do
app/models/miq_schedule_worker/runner.rb-151-      # Queue authentication checks for CIs with credentials
--
app/models/miq_schedule_worker/runner.rb-156-    every = worker_settings[:drift_state_purge_interval]
app/models/miq_schedule_worker/runner.rb:157:    scheduler.schedule_every(every, :first_in => every) do
app/models/miq_schedule_worker/runner.rb-158-      enqueue(:drift_state_purge_timer)
--
app/models/miq_schedule_worker/runner.rb-171-    job_not_found_delay = worker_settings[:evm_snapshot_delete_delay_for_job_not_found]
app/models/miq_schedule_worker/runner.rb:172:    scheduler.schedule_every(every, :first_in => every) do
app/models/miq_schedule_worker/runner.rb-173-      enqueue([:job_check_for_evm_snapshots, job_not_found_delay])
--
app/models/miq_schedule_worker/runner.rb-195-    every = worker_settings[:report_result_purge_interval]
app/models/miq_schedule_worker/runner.rb:196:    scheduler.schedule_every(every, :first_in => every) do
app/models/miq_schedule_worker/runner.rb-197-      enqueue(:miq_report_result_purge_timer)
--
app/models/miq_schedule_worker/runner.rb-200-    every = worker_settings[:container_entities_purge_interval]
app/models/miq_schedule_worker/runner.rb:201:    scheduler.schedule_every(every, :first_in => every) do
app/models/miq_schedule_worker/runner.rb-202-      enqueue(:archived_entities_purge_timer)
--
app/models/miq_schedule_worker/runner.rb-205-    every = worker_settings[:binary_blob_purge_interval]
app/models/miq_schedule_worker/runner.rb:206:    scheduler.schedule_every(every, :first_in => every) do
app/models/miq_schedule_worker/runner.rb-207-      enqueue(:binary_blob_purge_timer)
--
app/models/miq_schedule_worker/runner.rb-210-    every = worker_settings[:notifications_purge_interval]
app/models/miq_schedule_worker/runner.rb:211:    scheduler.schedule_every(every, :first_in => every) do
app/models/miq_schedule_worker/runner.rb-212-      enqueue(:notification_purge_timer)
--
app/models/miq_schedule_worker/runner.rb-215-    every = worker_settings[:task_purge_interval]
app/models/miq_schedule_worker/runner.rb:216:    scheduler.schedule_every(every, :first_in => every) do
app/models/miq_schedule_worker/runner.rb-217-      enqueue(:task_purge_timer)
--
app/models/miq_schedule_worker/runner.rb-220-    every = worker_settings[:vim_performance_states_purge_interval]
app/models/miq_schedule_worker/runner.rb:221:    scheduler.schedule_every(every, :first_in => every) do
app/models/miq_schedule_worker/runner.rb-222-      enqueue(:vim_performance_states_purge_timer)
--
app/models/miq_schedule_worker/runner.rb-237-    schedule_settings_for_ems_refresh.each do |klass, local_every|
app/models/miq_schedule_worker/runner.rb:238:      scheduler.schedule_every(every, :first_in => local_every) do
app/models/miq_schedule_worker/runner.rb-239-        enqueue([:ems_refresh_timer, klass])
--
app/models/miq_schedule_worker/runner.rb-253-    scheduler = scheduler_for(:scheduler)
app/models/miq_schedule_worker/runner.rb:254:    scheduler.schedule_every(every, :first_at => Time.current + 1.minute) do
app/models/miq_schedule_worker/runner.rb-255-      enqueue(:check_for_timed_out_active_tasks)
--
app/models/miq_schedule_worker/runner.rb-264-    scheduler = scheduler_for(:scheduler)
app/models/miq_schedule_worker/runner.rb:265:    scheduler.schedule_every(every, :first_at => time_at) do
app/models/miq_schedule_worker/runner.rb-266-      enqueue([:generate_chargeback_for_service, :report_source => "Daily scheduler"])
--
app/models/miq_schedule_worker/runner.rb-312-    scheduler.schedule_every(
app/models/miq_schedule_worker/runner.rb:313:      every,
app/models/miq_schedule_worker/runner.rb-314-      :first_in => first_in,
--
app/models/miq_schedule_worker/runner.rb-320-    scheduler.schedule_every(
app/models/miq_schedule_worker/runner.rb:321:      every,
app/models/miq_schedule_worker/runner.rb-322-      :first_in => first_in,
--
app/models/miq_schedule_worker/runner.rb-336-    scheduler.schedule_every(
app/models/miq_schedule_worker/runner.rb:337:      every,
app/models/miq_schedule_worker/runner.rb-338-      :first_in => first_in,
```

Reply:
I forget, did you test schedules with intervals coming from our settings.yml that were activesupport durations?  I think the answer is yes.  I vaguely recall seeing chargeback schedules that appear to use durations:

```
      :chargeback_generation_interval: 1.day
```

I'm asking here just to make sure it's in the PR and not just in chat.

Reply:
yup, it's queuing authentication_check_types